src/imbalances_script.py

```diff
@@ -0,0 +1,242 @@
+import requests
+import json
+from typing import Dict, List, Optional, Tuple
+from web3 import Web3
+from src.config import CHAIN_RPC_ENDPOINTS
+from src.constants import (SETTLEMENT_CONTRACT_ADDRESS, NATIVE_ETH_TOKEN_ADDRESS, 
+                           WETH_TOKEN_ADDRESS, SDAI_TOKEN_ADDRESS)
+
+EVENT_TOPICS = {
+    'Transfer': 'Transfer(address,address,uint256)',
+    'ERC20Transfer': 'ERC20Transfer(address,address,uint256)',
+    'WithdrawalWETH': 'Withdrawal(address,uint256)',
+    'DepositSDAI': 'Deposit(address,address,uint256,uint256)',
+    'WithdrawSDAI': 'Withdraw(address,address,address,uint256,uint256)',
+}
+
+WEB3_INSTANCES = {chain: Web3(Web3.HTTPProvider(url)) for chain, url in CHAIN_RPC_ENDPOINTS.items()}
+
+class RawTokenImbalances:
+    def __init__(self, web3: Web3, chain_name: str):
+        self.web3 = web3
+        self.chain_name = chain_name
+
+    def get_transaction_trace(self, transaction_hash: str, chain_url: str) -> list[dict] | None:
```

Comment:
I am a bit confused with this one. The RawTokenImbalances object contains a web3 variable. What you are doing here seems like a workaround, and i am not sure why this is needed.